driver/src/price_estimation/dexag/api.rs

```diff
@@ -80,16 +75,14 @@ impl DexagApi for DexagHttpApi {
             q.append_pair("dex", "ag");
         }
 
-        self.client
-            .get_async(url.to_string())
-            .map(|response| {
-                let mut body = response.context("failed to retrieve price from dexag")?;
-                let price = body
-                    .json::<Price>()
-                    .context("failed to parse price json from dexag")?;
-                Ok(price.price)
-            })
-            .boxed()
+        async move {
+            Ok(self
+                .client
+                .get_json_async(url.as_str())
+                .await
+                .context("failed to parse price json from dexag")?)
+        }
+        .boxed()
```

Comment:
There is a problem here.

This function here returns f64 so `get_json_async` tries to deserialize into f64 but it should deserialize into `Price`.

```
async move {
            Ok(self
                .client
                .get_json_async::<_, Price>(url.as_str())
                .await
                .context("failed to parse price json from dexag")?
                .price)
        }
        .boxed()
```

Reply:
That is the only problem I see.

Reply:
Fixed :)